### PR TITLE
Adding flag to retain “index.html” in results URLs

### PR DESCRIPTION
**Pull request by kenpetti-toasttab**
# What
* Add a new CLI flag to retain “index.html” in search result links
* Added a test case for this flag
* Added documentation with usage

# Why
My static site does not use “pretty” links. The current Pagefund behavior generates search result links that do not accurately link to my content. This change makes the links compatible.

# Testing
I tested this manually locally. I also added a simple test for this flag. Any ideas for additional test cases are welcome.

### CODE REVIEW

File: docs/content/docs/search-config.md [hunk 1]
```diff
@@ -63,6 +63,16 @@ Overrides the bundle directory. In most cases this should be automatically detec
...
+### Keep Index URL
+
+```json
+{
+    "keepIndexUrl": true
+}
+```
+
+Keeps `index.html` at the end of search result paths. For example, a file called "public/cat/index.html will have a search result url of `public/cat/index.html`. Defaults to `false`, stripping `index.html`.
+
```
Comment (bglw) [inline docs/content/docs/search-config.md:75]: These docs should actually live in `docs/config-options.md` — this page is for the browser-side config — but I can move them to the correct place after this is merged 🙂

**Review by bglw — APPROVED** [review]
Thanks for this, nice addition 🙂

**Comment by kenpetti-toasttab** [comment]
No problem! When should I expect this to be in a release?

**Comment by bglw** [comment]
A little while away from a stable release sorry @kenpetti-toasttab — I'm working on a raft of features that might push out to a full `v1.0.0` release.

I am pushing out some prereleases while testing, though, so if you're running through npx you can use:
```
npx pagefind@0.12.0-beta.0
```

And that release contains this PR 🙂 _(if you're using the binaries directly, [you can find them here](https://github.com/CloudCannon/pagefind-beta/releases/tag/v0.12.0-beta.0))_

These prereleases will eventually contain some breaking changes, but `0.12.0-beta.0` doesn't yet, so if you use that one directly you should be fine 🙂

**Comment by kenpetti-toasttab** [comment]
Great, thanks! Really happy so far, looking forward to seeing where this project goes!

**Comment by bglw** [comment]
Change of plans! [v0.12.0](https://github.com/CloudCannon/pagefind/releases/tag/v0.12.0) is now out containing this feature 🙂

**Pull request merged**